### PR TITLE
Feature/add skip functionality

### PR DESCRIPTION
**Pull request by jnation3406**
Adding all of you for visibility... 

- It uses tom_alertstreams and the alertstream_handler functions to ingest Messages from gcn classic over kafka and hop. 
- Models have been added for NonlocalizedEvents, NonlocalizedEventSequences, and Targets. Those constructs are created and linked as part of the parsing of LVC notices, LVC counterparts, and GCN Circulars. We will need to add more parsers for other topics/formats as we learn about them.
- The serializers, filters, and views are set up in a way to make querying for anything related to a nonlocalizedevent easy - everything can be filtered by event_id. Messages and Targets also support filtering by cone_search or polygon_search. 
- Only sample LVC Notices are coming through GCN classic over kafka right now, so there is a management command "inject_message" that can inject messages for the 3 main types we can parse right now in the format we expect them to be in. 
- I've changed the expected format for the hermes messages to put "event_id" within the data dict, and to move what was in the data dict one level deeper under "photometry" or "candidates" for those two hermes message types. There is also a hermes message parse which should correctly link the hermes data to Targets and NonLocalizedEvents in the db, but this is untested for now.
- We should not merge this until after Andy's demo, since this will require corresponding changes in the frontend and the tom datastreams stuff Joey has been working on for tom/hermes data sharing.

### CODE REVIEW

File: hermes/parsers/base_parser.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        filename = filename.replace('bayestar.fits.gz', 'bayestar.multiorder.fits')
+        return os.path.join(base, filename)
+
+
+class DefaultParser(BaseParser):
+
+    def __repr__(self):
+        return 'Default Parser'
+
+    def parse(self, message):
```
Comment (phycodurus) [inline hermes/parsers/base_parser.py:100]: The other parsers are returning a Boolean. Should this parser follow suit?

File: hermes/test/test_parsers.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+class TestLVCCounterpartParser(TestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.event_id = 'S123321'
+        message_text = get_lvc_notice_text(type='LVC_INITIAL', event_id=self.event_id )
+        self.message, _ = Message.objects.get_or_create(
+            topic='test_topic',
+            message_text=message_text
+        )
+        GCNLVCNoticeParser().parse(self.message)
```
Comment (phycodurus) [inline hermes/test/test_parsers.py:102]: I was looking to here to see how the return value of `parse()` was used. I guess I'm still confused about what the intention is with respect to the Boolean returned.

File: hermes/management/commands/hopingest.py [hunk 3]
```diff
@@ -1,444 +0,0 @@
...
-
-logger = logging.getLogger(__name__)
-#logger.setLevel(logging.DEBUG)
-
-class UpdateTopicsException(Exception):
-    pass
-
-SCIMMA_KAFKA_URL = 'kafka://kafka.scimma.org/'
-
-class Command(BaseCommand):
```
Comment (phycodurus) [inline hermes/management/commands/hopingest.py:26]: Replacing this management command with `tom-alertstreams` `readstreams` will propagate in to the `helm-chart/templates/deployment.yaml` where we start the management command in it's container. I'm happy to make that small change when the time comes.

File: hermes/tests.py [hunk 4]
```diff
@@ -1,3 +0,0 @@
-from django.test import TestCase
-
-# Create your tests here.
```
Comment (phycodurus) [inline hermes/tests.py:3]: Thank you, thank you, thank you for adding tests!!! (I'm so embarrassed that we've neglected testing until now!).

File: hermes/management/commands/inject_message.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        "number":"28609",
+        "subject":"{event_id}: No candidate counterparts from the Zwicky Transient Facility",
+        "date":"{published}",
+        "from":"{author}"
+    },
+    "body":"This is an injected test gcn circular message."
+}
+
+class Command(BaseCommand):
+    help = 'Inject a Message into the system and attempt to parse it, given a few fields being set to mimick different message types'
```
[the remark this answers is not in the document]
Comment (jnation3406) [reply hermes/management/commands/inject_message.py:26]: Yeah, it was for me to do integration testing, and I use it in the unit tests too. I like the readability of the _ in the command name better... I don't know why django mashes multiple words together with no delimiter.

File: hermes/parsers/base_parser.py (hunk 1 above)
Comment (jnation3406) [reply hermes/parsers/base_parser.py:100]: I'll just remove this - the way I have it implemented, we don't need a default parser implementation as a parser will only be used for known message formats.

File: hermes/management/commands/hopingest.py (hunk 3 above)
Comment (jnation3406) [reply hermes/management/commands/hopingest.py:26]: Okay, I've updated the helm charts command to be readstreams. We will also need to add the scimma user/pass and gcn user/pass env variables to the k8s secrets for hermes

Comment (phycodurus) [reply hermes/management/commands/hopingest.py:26]: I'll take care of this. 

And just a note to myself.:
* our current `HERMES_INGEST_USERNAME` should become `SCIMMA_AUTH_USERNAME`, and
* our current `HERMES_INGEST_PASSWORD` should become `SCIMMA_AUTH_PASSWORD`
* And we need to add `GCN_CLASSIC_CLIENT_ID` and `GCN_CLASSIC_CLIENT_SECRET`

**Review by phycodurus — APPROVED** [review]

**Comment by phycodurus** [comment]
This just came to mind: The `hopingest` management command was replaced by the `tom-alertstreams`-supplied `readstreams` management command. One of the handlers in `hopingest` listened to the `sys.heartbeat` topic and used that as a trigger to update the list of publicly readable topics to ingest (since that can be changing in SCiMMA Auth at any time). Is there a different mechanism for that in this PR or should we carry that handler over?

**Comment by jnation3406** [comment]
> This just came to mind: The `hopingest` management command was replaced by the `tom-alertstreams`-supplied `readstreams` management command. One of the handlers in `hopingest` listened to the `sys.heartbeat` topic and used that as a trigger to update the list of publicly readable topics to ingest (since that can be changing in SCiMMA Auth at any time). Is there a different mechanism for that in this PR or should we carry that handler over?

I actually thought new topics would be created rarely enough that we would restart the readstreams command to pick them up.  If we want it to happen automatically, I think we'll have to add it to the tom_alertstreams code. It shouldn't be too bad to check for public topics again every once in a while.

**Comment by phycodurus** [comment]
> > This just came to mind: The `hopingest` management command was replaced by the `tom-alertstreams`-supplied `readstreams` management command. One of the handlers in `hopingest` listened to the `sys.heartbeat` topic and used that as a trigger to update the list of publicly readable topics to ingest (since that can be changing in SCiMMA Auth at any time). Is there a different mechanism for that in this PR or should we carry that handler over?
> 
> I actually thought new topics would be created rarely enough that we would restart the readstreams command to pick them up. If we want it to happen automatically, I think we'll have to add it to the tom_alertstreams code. It shouldn't be too bad to check for public topics again every once in a while.
I think Hermes needs to keep itself up-to-date with respect to publicly_readable topics as they come and go. And, yes,, the difference between `readstreams` and what `hopigest` became is that `hopigest` could update it's topics dynamically as they changed in SCiMMA Auth, while at the moment `readstreams` relies on the the `ALERT_STREAMS` config dictionary. But, come to think of it, there is a mechanism (via [get_alert_streams(alert_stream_configs: list):](https://github.com/TOMToolkit/tom-alertstreams/blob/a50c2f2390c65e194a3e50ee58aba5c151de5a51/tom_alertstreams/alertstreams/alertstream.py#L23)) to provide a dynamic configuration dictionary, but that's not been used before.

File: requirements.txt [hunk 6]
```diff
@@ -1,13 +1,16 @@
...
 djangorestframework==3.13.1
+django-filter>=2.4
 psycopg2-binary==2.9.3
 django-bootstrap4==22.1
+fastavro>=1.7
 whitenoise==6.0.0
 lcogt-logging==0.3.2
 hop-client~=0.7
+healpy>=1.16
+tom_alertstreams==0.5.2
```
Comment (jchate6) [inline requirements.txt:11]: I only see `tom_alertstreams==0.5.1` on pypi. 
Has 0.5.2 been deployed yet?

Comment (jnation3406) [reply requirements.txt:11]: hmm your right... I thought I had deployed it but looks like the action didn't run on tag. I have another PR for it now so I'll just try again with version 0.5.3

**Pull request merged**